drivers/telescope/lx200classic.cpp

```diff
@@ -322,6 +365,190 @@ bool LX200Classic::saveConfigItems(FILE *fp)
+    horizontalPos.az = parkAz + 180;
+    if (horizontalPos.az > 360)
+        horizontalPos.az -= 360;
+    horizontalPos.alt = parkAlt;
+
+    ln_equ_posn equatorialPos;
+
+    ln_get_equ_from_hrz(&horizontalPos, &observer, ln_get_julian_from_sys(), &equatorialPos);
+
+    LOGF_DEBUG("Parking to RA (%f) DEC (%f)...", equatorialPos.ra / 15.0, equatorialPos.dec);
+    
+    //save the current AlignmentMode to UnparkAlignment
+    LX200Generic::getAlignment();
+    int curAlignment = IUFindOnSwitchIndex(&AlignmentSP);
+    IUResetSwitch(&UnparkAlignmentSP);
+    UnparkAlignmentS[curAlignment].s = ISS_ON;
+    UnparkAlignmentSP.s = IPS_OK;
+    IDSetSwitch(&UnparkAlignmentSP, nullptr);
+    saveConfig(true, UnparkAlignmentSP.name);
+
+    if (!Goto(equatorialPos.ra / 15.0, equatorialPos.dec))
+    {
+        ParkSP.s = IPS_ALERT;
+        IDSetSwitch(&ParkSP, "Parking Failed.");
+        return false;
+    }
+
+    EqNP.s     = IPS_BUSY;
+    TrackState = SCOPE_PARKING;
+    LOG_INFO("Parking is in progress...");
+
+    return true;
+}
+
+bool LX200Classic::UnPark()
+{
+    if (isSimulation() == false)
+    {
+        // Parked in Land alignment. Restore previous mode.
+        if (setAlignmentMode(PortFD, IUFindOnSwitchIndex(&UnparkAlignmentSP)) < 0)
+        {
+            LOG_ERROR("UnParking Failed.");
+            return false;
+        }
+        //Update the UI
+        LX200Generic::getAlignment();
+    }
+
+    // Then we sync with to our last stored position
+    double parkAz  = GetAxis1Park();
+    double parkAlt = GetAxis2Park();
+
+    char AzStr[16], AltStr[16];
+    fs_sexa(AzStr, parkAz, 2, 3600);
+    fs_sexa(AltStr, parkAlt, 2, 3600);
+    LOGF_DEBUG("Syncing to parked coordinates Az (%s) Alt (%s)...", AzStr, AltStr);
+
+    ln_lnlat_posn observer;
+    observer.lat = LocationN[LOCATION_LATITUDE].value;
+    observer.lng = LocationN[LOCATION_LONGITUDE].value;
+    if (observer.lng > 180)
+        observer.lng -= 360;
+
+    ln_hrz_posn horizontalPos;
+    // Libnova south = 0, west = 90, north = 180, east = 270
+
+    horizontalPos.az = parkAz + 180;
+    if (horizontalPos.az > 360)
+        horizontalPos.az -= 360;
+    horizontalPos.alt = parkAlt;
+
+    ln_equ_posn equatorialPos;
+
+    ln_get_equ_from_hrz(&horizontalPos, &observer, ln_get_julian_from_sys(), &equatorialPos);
+
+    if (isSimulation())
+    {
+        currentRA = equatorialPos.ra / 15.0;
+        currentDEC= equatorialPos.dec;
+    }
+    else
+    {
+        if ((setObjectRA(PortFD, equatorialPos.ra / 15.0) < 0) || (setObjectDEC(PortFD, equatorialPos.dec) < 0))
+        {
+            LOG_ERROR("Error setting Unpark RA/Dec.");
+            return false;
+        }
+
+        char syncString[256];
+        if (::Sync(PortFD, syncString) < 0)
+        {
+            LOG_WARN("Unpark Sync failed.");
+            return false;
+        }
+    }
+
+    SetParked(false);
+    return true;
+}
+
+bool LX200Classic::SetCurrentPark()
+{
+    ln_hrz_posn horizontalPos;
```

Comment:
Most of this is boilerplate copied from... lx200telescope? Is here any way for us to elevate this code so both of us can use it without copy pasta these entire methods?

Reply:
I took it from here: https://github.com/indilib/indi/blob/9f37544774c0cd22fe4289d60a3ff32a9711ca81/drivers/telescope/lx200ss2000pc.cpp#L329

I would not call it boiler plate. It converts AltAz parking coordinates to RaDec. And in one case the other way. The LX200 classic does not support AltAz position setting. Most later LX200 scopes do. Hence the need to control it in RaDec.  

Would there be a more compact helper function anywhere else?

Reply:
If there’s not already a “convertAltAz( &RA, &Dec )” and “convertRADec( &Alt, &Az)”, there should be 😄 but that’s like indibase type stuff

Reply:
I have moved the coordinates system conversion code to two separate methods. Those two methods could be moved down the specialization chain, or two a separate helper module.